include/envoy/network/dns.h

```diff
@@ -41,11 +43,14 @@ class DnsResolver {
   /**
    * Initiate an async DNS resolution.
    * @param dns_name supplies the DNS name to lookup.
+   * @param dns_lookup_family the DNS IP version lookup policy.
    * @param callback supplies the callback to invoke when the resolution is complete.
    * @return if non-null, a handle that can be used to cancel the resolution.
    *         This is only valid until the invocation of callback or ~DnsResolver().
    */
-  virtual ActiveDnsQuery* resolve(const std::string& dns_name, ResolveCb callback) PURE;
+  virtual ActiveDnsQuery* resolve(const std::string& dns_name,
+                                  const DnsLookupFamily& dns_lookup_family,
```

Comment:
I would just pass this by value, it might even be more efficient to do so (it's just a small integer).